certora/specs/HashConsensus.spec

```diff
@@ -0,0 +1,742 @@
+    assert slot > max_uint64 => lastReverted;
+}
+
+
+// 16. submitReport() - slot != refSlot of current frame => revert
+// Status: Pass
+// https://prover.certora.com/output/80942/ec099c0e1f2145ed9c91c9865aa62c78?anonymousKey=51a7cd4d068a5b22043288a52b7a2e2521585669
+rule cannotSubmitReportWhenSlotDoesNotMatchCurrentRefSlot() {
+    env e;
+    uint256 currentRefSlot; uint256 reportProcessingDeadlineSlot;
+    currentRefSlot, reportProcessingDeadlineSlot = getCurrentFrame(e);
+    
+    uint256 slot; bytes32 report; uint256 consensusVersion;
+    submitReport@withrevert(e, slot, report, consensusVersion);
+    assert slot != currentRefSlot => lastReverted;
+}
+
+
+// 17. submitReport() - same member cannot submit two different reports for the same slot
+// Status: Pass
+// https://prover.certora.com/output/80942/f74b693ec24c433cae80f54f8623ae78/?anonymousKey=30f5809f7e82a6cd9782fdc9947590d29242be4c
+rule sameMemberCannotSubmitDifferentReportForTheSameSlot() {
+    env e; env e2;
+
+    uint64 lastReportRefSlot; uint64 lastConsensusRefSlot; uint64 lastConsensusVariantIndex;
+    lastReportRefSlot, lastConsensusRefSlot, lastConsensusVariantIndex = helper_getReportingState(e);
+
+    uint256 lastProcessingRefSlot = helper_getLastProcessingRefSlot(e);
+
+    uint256 slot; bytes32 report; uint256 consensusVersion;
+    require slot > lastProcessingRefSlot;  // otherwise the report doesn't matter
+    require slot > lastConsensusRefSlot;  // reporting for a slot without consensus
+    
+    submitReport(e, slot, report, consensusVersion);
+
+    bytes32 report2;
+    submitReport@withrevert(e2, slot, report2, consensusVersion);
+
+    assert (ghostReportHash != 0) && (e2.msg.sender == e.msg.sender) && (report != report2) => lastReverted;
+}
+
+
+// 18. submitReport() - reportHash == 0 => revert
+// Status: Pass
+// https://prover.certora.com/output/80942/de955bfeccf045249d7470130ae3338d/?anonymousKey=bec48e627f110948e49b36201dad6c3d72622b0c
+rule cannotSubmitReportWithEmptyHash() {
+    env e;
+    uint256 slot; bytes32 report; uint256 consensusVersion;
+    submitReport@withrevert(e, slot, report, consensusVersion);
+    bool callReverted = lastReverted;
+
+    assert (report == ZERO_HASH()) => callReverted;
+}
+
+
+// 19. submitReport() - consensusVersion != getConsensusVersion() => revert
+// Status: Pass
+// https://prover.certora.com/output/80942/3beb82cf292e4db8b39f8a70534cc8fa/?anonymousKey=2665d9fab09defd5f246f302ef5da1e77c058379
+rule submitReportMustHaveCorrectConsensusVersion() {
+    env e;
+    uint256 slot; bytes32 report; uint256 consensusVersion;
+    uint correctConsensusVersion = helper_getConsensusVersion(e);
+    submitReport@withrevert(e, slot, report, consensusVersion);
+
+    assert (consensusVersion != correctConsensusVersion) => lastReverted;
+}
+
+
+// 20. submitReport() - verify that the deadline (frame.reportProcessingDeadlineSlot) is explicitly ==  refSlot + FrameSize
```

Comment:
Examining the counter example, I think you need to add some requirement on the maximum value of the refSlot, it looks too big.